source/code-snippets/usage-examples/changeStream.js

```diff
@@ -21,21 +26,15 @@ async function run() {
       console.log("received a change to the collection: \t", next);
     });
 
-    // use a timeout to ensure the listener is registered before the insertOne
-    // operation is called.
-    await new Promise(resolve => {
-      setTimeout(async () => {
-        await movies.insertOne({
-          test: "sample movie document",
-        });
-        // wait to close `changeStream` after the listener receives the event
-        setTimeout(async () => {
-          resolve(
-            await changeStream.close(() => console.log("closed the change stream"))
-          );
-        }, 1000);
-      }, 1000);
+    await simulateAsyncPause();
```

Comment:
While this code snippet is definitely more readable than what was here before (great job!), I think this doesn't substantively change the nature of the example (it still waits, inserts a document, waits, closes the change stream). This [is the comment](https://github.com/mongodb/docs-node/pull/228#discussion_r686283833) that initiated this ticket. I think this is mainly about how the example seems largely artificial in addition to how the example doesn't take advantage of helpful promise syntax.

Could this code snippet be updated to more closely match a real world use case for change streams? [I think you did this beautifully in the go docs](https://github.com/mongodb/docs-golang/blob/master/source/usage-examples/watch.txt#L30-L33).

Reply:
I think this code snippet could be used to mirror what is in the go docs: 

    import { MongoClient } from "mongodb";
    
    // Replace the uri string with your MongoDB deployment's connection string.
    const uri = "<your uri>";
    
    const client = new MongoClient(uri);
    
    var changeStream;
    async function run() {
      try {
        await client.connect();
        const database = client.db("insertDB");
        const movies = database.collection("foods");
    
        // open a Change Stream on the "movies" collection
        changeStream = movies.watch();
    
        console.log("Waiting for changes...");
        while (changeStream.hasNext()) {
          const change = await changeStream.next();
          console.log("received a change to the collection: \t", change);
        }
      } finally {
        await changeStream.close(() => console.log("closed the change stream"));
        await client.close();
      }
    }
    run().catch(console.dir);